src/components/UnlockWallet/index.tsx

```diff
@@ -30,6 +31,49 @@ function warnUser(title: string, message?: string | undefined): void {
     message,
   });
 }
+function changeId(chainName : any) {
+  var chains =  JSON.parse(JSON.stringify(internalAddressesPerNetwork))
+  const chain = chains[chainName]
+  try {
+    switch(chain.chainId){
+      // If the chain is default to Metamask if will use wallet_switchEthereumChain
+      case 4:
+        window.ethereum.request({
+          id: 1,
+          jsonrpc: '2.0',
+          method: 'wallet_switchEthereumChain',
+          params: [
+            {
+              chainId: '0x4',
```

Comment:
Constant chain ID? Why 4 specifically?

Reply:
I switched this to `
        chainId: "0x" + chainId.toString(16)`, but since Rinkeby is part of the default chains that Metamask has, it will call `wallet_switchEthereumChain `. Unless other default chains are added to Agave, the only chain that will need this request is Rinkeby.